### PR TITLE
Backports and changelog for 0.12.1

### PR DESCRIPTION
**Pull request by heartsucker**
## Status

Ready for review

## Description of Changes

Backports the currently closed tickets of the 0.12.1 milestone into the 0.12.0 release branch. Updates the changelog.

### CODE REVIEW

File: changelog.md [hunk 1]
```diff
@@ -1,5 +1,20 @@
...
 
+## 0.12.1~rc1
+
+### Web Application
+
+* Add "Back to submission page" link to NoScript docs (#4208)
+
+### Operations
+
+* Ensured `gcc` and WiFi related packages are not installed on Xenial on upgrade (#4163)
```
Comment (emkll) [inline changelog.md]: `gcc` will still installed per: https://github.com/freedomofpress/securedrop/issues/4163#issuecomment-468681362 , apologies should have changed the issue name.

Comment (heartsucker) [reply changelog.md]: Ok, so this is still only issue 4163 and the change would be ```s/`gcc` and//``` basically yeah?

Comment (emkll) [reply changelog.md]: yup!

File: admin/securedrop_admin/__init__.py [hunk 2]
```diff
@@ -642,7 +642,7 @@ def check_for_updates(args):
 
 
 def get_release_key_from_keyserver(args, keyserver=None, timeout=45):
-    gpg_recv = ['timeout', str(timeout), 'gpg', '--recv-key']
+    gpg_recv = ['timeout', str(timeout), 'gpg', '--batch', '--no-tty', '--recv-key']
```
Comment (emkll) [inline admin/securedrop_admin/__init__.py]: it seems like the `-x` option was not used when cherry-picking this commit?

Comment (heartsucker) [reply admin/securedrop_admin/__init__.py]: This wasn't a cherry-pick. I had to add this to get the lint to pass.

**Pull request merged**